apps/settings/test/unit/panels/developer/developer_test.js

```diff
@@ -0,0 +1,84 @@
+'use strict';
+
+/* globals loadBodyHTML*/
+
+requireApp('settings/shared/test/unit/load_body_html_helper.js');
+
+suite('Developer > ', function() {
+
+  var modules = [
+    'MockDialogService',
+    'unit/mock_apps_cache',
+    'shared_mocks/mock_screen_layout',
+    'unit/mock_settings_cache',
+    'shared_mocks/mock_navigator_moz_power',
+    'panels/developer/developer'
+  ];
+  var maps = {
+    'panels/developer/developer': {
+      'modules/dialog_service': 'MockDialogService',
+      'modules/apps_cache': 'unit/mock_apps_cache',
+      'shared/screen_layout': 'shared_mocks/mock_screen_layout',
+      'modules/settings_cache': 'unit/mock_settings_cache'
+    }
+  };
+  var developer, realMozPower, mozPower;
+
+  suiteSetup(function(done) {
+    loadBodyHTML('_developer.html');
+
+    realMozPower = navigator.mozPower;
+
+    // Create a new requirejs context
+    var requireCtx = testRequire([], maps, function() {});
+
+    // Define MockDialogService
+    define('MockDialogService', function() {
+      return {
+        alert: function() {},
+        confirm: function() {}
+      };
+    });
+
+    requireCtx(modules, function(MockDialogService, MockAppsCache,
+      MockScreenLayout, MockSettingsCache, MockNavigatorMozPower, Developer) {
+      navigator.mozPower = mozPower = MockNavigatorMozPower;
+      var panel = document.body;
+      developer = Developer();
+      developer.init({
+        resetSwitch: panel.querySelector('.reset-devtools'),
+        ftuLauncher: panel.querySelector('.ftuLauncher'),
+        softwareHomeButton: panel.querySelector('.software-home-button'),
+        homegesture: panel.querySelector('.homegesture')
+      });
+      done();
+    });
+  });
+
+  suiteTeardown(function() {
+    navigator.mozPower = realMozPower;
+  });
+
+  suite('developer initialized', function() {
+    test('constructed', function() {
+      assert.isFalse(developer._elements.resetSwitch.disabled);
+    });
+  });
+
+  suite('factory reset', function() {
+    test('button clicked', function() {
+      var _resetDeviceStub = sinon.stub(developer, '_resetDevice');
+      developer._elements.resetSwitch.click();
+      assert(_resetDeviceStub.calledOnce);
+      _resetDeviceStub.restore();
+    });
+
+    test('_wipe', function() {
+      sinon.stub(mozPower, 'factoryReset');
+      ['normal', 'root'].forEach(reason => {
+        developer._wipe(reason);
+        assert(mozPower.factoryReset.calledWith(reason));
+      });
```

Comment:
We should add a test ensuring we do the correct type of reset based on the settings value.